### ISSUE TITLE
Some 'deep' examples on the chai webpage arn't working

### ISSUE THREAD

**Issue by ShanWehttam** [chaijs/chai#1300]
Hi there,
Using "version": "4.2.0"
        "assertion-error": "^1.1.0",
        "check-error": "^1.0.2",
        "deep-eql": "^3.0.1",
        "get-func-name": "^2.0.0",
        "pathval": "^1.1.0",
        "type-detect": "^4.0.5"

Some of the '.deep' 'expect' assertions that are given as examples on the chai website arn't working, and i wondered if it was an issue others are experiencing.  Are they supposed to fail?

expect({'.a': {'[b]': 'x'}}).to.have.nested.property('\\.a.\\[b\\]'); 
  // TypeError: Cannot read property 'property' of undefined
expect({a: {b: ['x', 'y']}}).to.have.nested.property('a.b[1]'); 
  // TypeError: Cannot read property 'property' of undefined
expect({x: {a: 1}}).to.have.deep.property('x', {a: 1}); 
  // AssertionError: expected { x: { a: 1 } } to have a deep property 'x' of { a: 1 }, but got { a: 1 }
expect({x: {a: 1}}).to.deep.include({x: {a: 1}}); 
  // AssertionError: expected { x: { a: 1 } } to have a property 'x' of { a: 1 }, but got { a: 1 }
expect(new Set([{a: 1}])).to.have.deep.keys([{a: 1}]); 
 //AssertionError: expected {} to have key { a: 1 } 

There are probably more out there, but i'd appreciate some assistance to better my understanding of chai, thanks

**Comment by megakoresh** [comment]
Yes, can confirm on 4.2.0 the docs are misleading. Nested inclusion of a subset does not work.

**Comment by mercmobily** [comment]
> Nested inclusion of a subset does not work.

Is there any plan to make it work?
It's a pretty common use case, where the response has an inner-inner object somewhere and the test needs to check if a subset of properties are there.

OK, maybe not as common as I hope, but... hopefully common enough? :D
